internal/k8s/controller.go

```diff
@@ -976,8 +992,49 @@ func (lbc *LoadBalancerController) syncVirtualServer(task task) {
 				glog.Errorf("Error when updating the status for VirtualServerRoute %v/%v: %v", vsr.Namespace, vsr.Name, err)
 			}
 		}
+
+		vsrkey := fmt.Sprintf("%s/%s", vsr.Namespace, vsr.Name)
+		for i, r := range vsrlist {
+			key := fmt.Sprintf("%s/%s", r.Namespace, r.Name)
+			if key == vsrkey {
+				if len(vsrlist) > 1 {
+					vsrNotRef = append(vsrNotRef[:i-vsrRemoved], vsrNotRef[(i-vsrRemoved)+1:]...)
+					vsrRemoved++
+				} else {
+					vsrNotRef = []*conf_v1.VirtualServerRoute{}
+				}
+			}
+		}
+	}
+	reason := "Ignored"
+	for _, vsr := range vsrNotRef {
+		msg := fmt.Sprintf("Ignored by VirtualServer %v/%v", vs.Namespace, vs.Name)
+		lbc.recorder.Eventf(vsr, api_v1.EventTypeWarning, "Ignored", msg)
+		lbc.statusUpdater.UpdateVirtualServerRouteStatus(vsr, conf_v1.StateInvalid, reason, msg)
 	}
+}
 
+func findAllVirtualServerRoutesForVirtualServer(vs *conf_v1.VirtualServer, vsrs []*conf_v1.VirtualServerRoute) []*conf_v1.VirtualServerRoute {
+	var result []*conf_v1.VirtualServerRoute
+	key := fmt.Sprintf("%s/%s", vs.Namespace, vs.Name)
+	for _, r := range vsrs {
+		if r.Status.ReferencedBy == key {
+			result = append(result, r)
+		}
+	}
+	return result
+}
+
+func findVirtualServerRoutesForVirtualServer(virtualserver *conf_v1.VirtualServer, virtualServerRoutes []*conf_v1.VirtualServerRoute) []*conf_v1.VirtualServerRoute {
```

Comment:
Lets keep func args short and consistent?

```suggestion
func findVirtualServerRoutesForVirtualServer(vs *conf_v1.VirtualServer, vsrs []*conf_v1.VirtualServerRoute) []*conf_v1.VirtualServerRoute {
```